baseapp/abci.go

```diff
@@ -707,25 +675,27 @@ func (app *BaseApp) GetBlockRetentionHeight(commitHeight int64) int64 {
 		retentionHeight = commitHeight - cp.Evidence.MaxAgeNumBlocks
 	}
 
-	// Define the state pruning offset, i.e. the block offset at which the
-	// underlying logical database is persisted to disk.
-	statePruningOffset := int64(app.cms.GetPruning().KeepEvery)
-	if statePruningOffset > 0 {
-		if commitHeight > statePruningOffset {
-			v := commitHeight - (commitHeight % statePruningOffset)
-			retentionHeight = minNonZero(retentionHeight, v)
-		} else {
-			// Hitting this case means we have persisting enabled but have yet to reach
-			// a height in which we persist state, so we return zero regardless of other
-			// conditions. Otherwise, we could end up pruning blocks without having
-			// any state committed to disk.
-			return 0
+	if app.snapshotManager != nil {
```

Comment:
When can snapshot manager be nil? Seems like it has to be set for pruning to work at all. Should we instead guard that snapshot manager is not nil?

Reply:
It can be nil when we don't set snapshot options.

E.g: https://github.com/osmosis-labs/cosmos-sdk/blob/33fb68b22590de37a7e45c7d64cbf65a9bf074ac/baseapp/baseapp_test.go#L2268

The pruning works without snapshots options / manager being set. In `pruning.Manager` we have [`snapshotInterval`](https://github.com/osmosis-labs/cosmos-sdk/blob/33fb68b22590de37a7e45c7d64cbf65a9bf074ac/pruning/manager.go#L94) which is 0 by default. When it is 0, we prune all heights right away. If the snapshot manager / options / interval are set, then the snapshot heights are skipped.

Reply:
Ohh this only matters for retention height, sorry, this logic confused me for a bit. Didnt realize we were in `func (app *BaseApp) GetBlockRetentionHeight`.